service/worker/replicator/replicationTask.go

```diff
@@ -515,6 +554,11 @@ func (t *workflowReplicationTask) Nack() {
 	t.metricsClient.RecordTimer(t.metricsScope, metrics.ReplicatorLatency, t.timeSource.Now().Sub(t.startTime))
 
 	t.state = task.TaskStateNacked
+
+	// TODO: need to discuss whether it is the good place to place the counter increment for the metrics because
```

Comment:
Sorry Wen,
Instead of emitting metrics, can we just add a log here with debugging info such as domainID, workflowID, runID, taskID?
It will still helpful for quick troubleshooting, but avoid passing domainCache everywhere and tag every replication tasks?
What your option @venkat1109 @meiliang86 ?